apps/charterafrica/src/lib/ecosystem/github/processData.js

```diff
@@ -76,16 +76,17 @@ async function fetchGithubApi(path, tag) {
   };
   try {
     const res = await fetch(url, { headers });
-    const response = await res.json();
-    const eTag = res.headers.get("ETag") || "";
     if (res.ok) {
+      const response = await res.json();
+      const eTag = res.headers.get("ETag");
       return { ...response, eTag };
     }
     if (res.status !== 304) {
+      const response = await res.json();
```

Comment:
Where are we using this `response`?

Reply:
We are only posting it to Sentry

Reply:
I don't see where we're actually posting it to Sentry @koechkevin. We're capturing error and sending it to Sentry but error includes `res`, not `response`.

Reply:
Response stringified in [message](https://github.com/CodeForAfrica/ui/blob/1e3e064db83f70d1fec21bfa3b7e372971f2347b/apps/charterafrica/src/lib/ecosystem/github/processData.js#L87)

Reply:
Cool, cool. For my understanding, what was the actual issue with the old approach?

Reply:
@kilemensi [Unexpected JSON format error occurs when an eTag is provided and github status code is 304.](https://code-for-africa.sentry.io/issues/4397875177/?project=4505192814084096&query=is%3Aunresolved+level%3Aerror&referrer=issue-stream&statsPeriod=14d&stream_index=3).

This happened because we had `res.json()` before checking on response status. So to fix this, we only have to move `res.json()` inside  if block